ntbs-service/DataAccess/AlertRepository.cs

```diff
@@ -193,6 +194,18 @@ public async Task CloseUnmatchedLabResultAlertsForSpecimenIdAsync(string specime
             await CloseAlertRangeAsync(alertsToClose);
         }
 
+        public async Task CloseUnmatchedLabResultAlertForSpecimenAndNotificationAsync(string specimenId,
+            int notificationId)
+        {
+            var alertToClose = await _context.Alert.OfType<UnmatchedLabResultAlert>()
+                .SingleAsync(alert => alert.AlertStatus == AlertStatus.Open

```

Comment:
It seems a bit risky to use `SingleAsync` here... do we know for certain that the alert will exist, and will still be Open?

Reply:
No you're right, we do not for definite know there is an open alert. Changed now